espressif/idf-eclipse-plugin
# IEP-955 Refactoring SerialFlashLaunchConfigDelegate class

## Description

summary:
* Added openocd_path, openocd_exe, OPENOCD_SCRIPTS variable and resolver for these variables
* Moved responsibility of parsing dynamic variables from SerialFlashLaunchConfigDelegate to the variable resolver
* Added openocd prefix to the arguments line in the CMakeMainTab2 so the default value looks like this:
`${openocd_path}\${openocd_exe} -s ${OPENOCD_SCRIPTS} -f board/esp32-wrover-kit-1.8v.cfg`
* Added argument line check method in the SerialFlashLaunchConfigDelegate to provide backward compatibility for the use case, when `${openocd_path}\${openocd_exe}` prefix is missing from the argument line (Previous behavior) 
* Moved flash over jtag to the ESPFlashUtil class 
* Added tests for variable resolver

Fixes # ([IEP-955](https://jira.espressif.com:8443/browse/IEP-955))

## Type of change

Please delete options that are not relevant.
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

## How has this been tested?
Test 1:
- create a new project 
- modify launch configuration -> select `flash over JTAG`
- make sure that now the arguments line looks like this `${openocd_path}\${openocd_exe} -s ${OPENOCD_SCRIPTS} -f board/esp32-wrover-kit-1.8v.cfg`

Test 2:
- try flash over jtag with the default argument line
Test 3:
- try flash over jtag with modified argument line (remove ${openocd_path}\${openocd_exe} part) - should be working and proving backward compatibility 



**Test Configuration**:
* ESP-IDF Version:
* OS (Windows,Linux and macOS):

## Dependent components impacted by this PR:

- Argument line of the launch configuration with flash over jtag option
- flashing

## Checklist
